src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/RetryExceptionPolicyFactory.cs

```diff
@@ -54,16 +62,60 @@ public AsyncPolicy RetryPolicy
             }
         }
 
-        private static AsyncRetryPolicy CreateExtendedRetryPolicy(int maxRetries, int maxWaitTimeInSeconds)
+        public AsyncPolicy BackgroundWorkerRetryPolicy => _backgroundJobRetryPolicy;
+
+        private AsyncRetryPolicy CreateExtendedRetryPolicy(int maxRetries, int maxWaitTimeInSeconds, bool useExponentialRetry = false)
         {
+            // Define a sleep duration provider based on the retry strategy
+            TimeSpan SleepDurationProvider(int retryAttempt, Exception exception)
+            {
+                // Respect x-ms-retry-after-ms from RequestRateExceededException
+                if (exception.AsRequestRateExceeded()?.RetryAfter is TimeSpan retryAfter)
+                {
+                    return retryAfter;
+                }
+
+                // Respect x-ms-retry-after-ms from CosmosException
+                if (exception is CosmosException cosmosException && cosmosException.StatusCode == HttpStatusCode.TooManyRequests && cosmosException.RetryAfter.HasValue)
+                {
+                    return cosmosException.RetryAfter.Value;
+                }
+
+                // Exponential backoff is used for background jobs. Given current values, exponential backoff is used for the first 10 retries. After that a fixed wait time of_exponentialMaxDelayMs (60 seconds) is used.
+                // Jitter is multiplied by the retry attempt to increase the randomness of the retry interval for longer retry delays (especially retry > 10).
+                if (useExponentialRetry)
+                {
+                    // Calculate exponential backoff with a cap of 60 seconds
+                    var backoff = Math.Min(Math.Pow(2, retryAttempt) * _exponentialBackoffBaseDelayMs, _exponentialMaxDelayMs);
+
+                    var jitter = RandomNumberGenerator.GetInt32(0, _exponentialMaxJitterMs) * retryAttempt; // Add jitter in milliseconds
+                    return TimeSpan.FromMilliseconds(backoff + jitter);
+                }
+
+                // Default fixed wait time
+                return TimeSpan.FromSeconds(2);
```

Comment:
Isn't two seconds too long for a default wait time? 
Could we adopt something lower? 500ms?

Reply:
2 seconds was the previous behavior for Cosmos Clients posting bundles (and background jobs). Since bundles are client operations, I would be a fan of logic like below - thoughts?

- Default time = 500ms + retryAttempt * jitter.

Our current config values will result in 6 retries, so this would result in a max retry delay of 2.5 seconds

Reply:
I like that path. Thanks!

Reply:
Fixed 😸